sycl/test/check_device_code/id_queries_fit_int.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -Xclang -fsycl-is-host -fsycl-id-queries-fit-in-int -O0 -c -S -emit-llvm -o %t.ll %s
+// RUN: %clangxx -fsycl -fsycl-id-queries-fit-in-int -O0 -c -S -emit-llvm -o %t.ll %s
```

Comment:
It looks like this test is supposed to check LLVM IR emitted by the host compiler. Right?
1. Should we remove  `-fsycl` as well? Is there a better way to get the result of host-side compilation?
2. (not relevant to this PR) Should it be located in sycl/test/**check_device_code**/ directory, if it doesn't check device code at all?

Reply:
It was unclear what this test was trying to accomplish with that RUN line. `-fsycl` kicks off a device and a host build, so it wasn't obvious what was intended when running in host-and-device-at-the-same-time mode (which isn't supported).

If the test is only interested in host mode functionality, it should remove `-fsycl` and pass `-Xclang -fsycl-is-host`.

Reply:
> It looks like this test is supposed to check LLVM IR emitted by the host compiler. Right?

Exactly.

Reply:
> By the way, what's the purpose of this file change?

Passing `-fsycl` to the Clang driver spins off two frontend executions, one that passes `-fsycl-is-device` and one that passes `-fsycl-is-host`. The frontend can only ever be in one of those modes at a time (it's now an error to try to pass both flags). This test was passing `-fsycl -Xclang -fsycl-is-host`, so for the host execution everything was fine, but for the device execution would end up in both device and host mode.

Reply:
So then this is not a host-only test?

Reply:
@AaronBallman, driver doesn't add paths to DPC++ headers implicitly w/o `-fsycl` option, so we need either add `-I <path to DPC++ headers>` (there should be examples of this in other tests) or probably better option is to update the test and run it in "device-only" mode (I guess this mode implies `-fsycl` and driver add path).

Reply:
I'd prefer not to try rewriting the test (it seems unrelated to the changes in this PR -- the test was already questionable). I'll try adding the -I path to correct the issue, thanks for the suggestion!

Reply:
I agree. The test already had this flow, so it's better to address it separately.

Reply:
It seems that passing the path was insufficient to correct the test. Given that the test is already in a broken state, I'd prefer if someone with more domain knowledge handled correcting the test deficiencies in a separate PR (so I would back to the original changes to the test, which were originally passing).

Reply:
> To validate what @s-kanaev says is intended we need to re-write the code to make it device code instead.

Then, there must be an issue for us to not forget about it.

Reply:
@s-kanaev -- I filed https://github.com/intel/llvm/issues/3335 to track this.

Reply:
Great thanks!